### ISSUE TITLE
[FEATURE] Add GDI text alpha blending fix

### ISSUE THREAD

**Issue by Undisputed00x** [snowie2000/mactype#1000]
Hi there, could you add a feature that fixes text alpha on GDI text, this issue is visible when having Acrylic blur windows by using [MicaForEveryone](https://github.com/MicaForEveryone/MicaForEveryone) with [DwmExtendFrameIntoClientArea function ](https://learn.microsoft.com/en-us/windows/win32/api/dwmapi/nf-dwmapi-dwmextendframeintoclientarea). Enabling DirectWrite doesn't fix the problem. Here some articles that discuss this problem [1](https://theartofdev.com/2013/10/24/transparent-text-rendering-with-gdi/)  [2](https://devblogs.microsoft.com/oldnewthing/20210915-00/?p=105687).
This project [ExplorerBlurMica](https://github.com/Maplespe/ExplorerBlurMica) that enables custom windows effects including acrylic blur has fixed this problem [here](https://github.com/Maplespe/ExplorerBlurMica/blob/b00889867f46910ffde52052b32b991f4b476934/ExplorerBlurMica/HookDef.cpp)

<details>
<summary>

 ### Examples of problem
</summary>
<p>

https://github.com/user-attachments/assets/932bf04a-9761-4b95-936d-8948c514f8b5

![SPOILER_PROBLEM1](https://github.com/user-attachments/assets/f030dce2-d92a-4759-a5fd-f5a599b715a8)
![SPOILER_PROBLEM_2](https://github.com/user-attachments/assets/6efe3468-2006-4002-b0c6-70f734dc653d)
</p>
</details> 

<details>
<summary>

### Fix of problem
</summary>
<p>

![SPOILER_EBM](https://github.com/user-attachments/assets/0624be8d-c240-45cf-83ae-05306a4f2bb6)
</p>
</details>

**Comment by snowie2000** [comment]
The standard Windows GDI drawtext call sets the alpha bit to 0 if the text color (a.k.a the painter color) is black or leaves whatever alpha bit is for all other colors. MacType follows the same rule to mantain max compatibility.

If you got the same result without mactype and needs a third-party app to "fix" the problem, that's a problem of the Windows by design and shouldn't be handled by mactype.

**Comment by snowie2000** [comment]
As of the Alpha bit problem in GlassFrame windows, I know it long ago. MacType has the ability to change alpha bit and was deliberately not touching it:
https://github.com/snowie2000/mactype/blob/c3a52fc5343641d5e91841674380c805edbbdcdd/ft.cpp#L387-L403

**Comment by Undisputed00x** [comment]
> If you got the same result without mactype and needs a third-party app to "fix" the problem, that's a problem of the Windows by design and shouldn't be handled by mactype.

Yeah, i know that's Windows own problem but i can't find any third party project that is closer to make a change on this issue systemwide other than mactype.




> As of the Alpha bit problem in GlassFrame windows, I know it long ago. MacType has the ability to change alpha bit and was deliberately not touching it:
> 
> https://github.com/snowie2000/mactype/blob/c3a52fc5343641d5e91841674380c805edbbdcdd/ft.cpp#L387-L403

If i set `255` on alpha variables of that fuction is it going to make text opaque on glass background ?

**EDIT**: uhm if it's not a problem for you can you share a compiled version with the alpha changes here to test ?😅

**Comment by Undisputed00x** [comment]
Any update on this 🙂?

**Comment by snowie2000** [comment]
It can be done easily. The problem is that I don't see a reason to make it to everyone.

**Comment by Undisputed00x** [comment]
I understand, but fixes a current issue with GDI text being invisible when HDR/10-bit colors are enabled with a combination of translucent background effect (mica/acrylic/blur) extending in the window client area where GDI text is painted. Im newbie at coding and find it hard to build the dependencies of the project 😅

<details><summary>List of tickets from other third party projects about this "problem"</summary>
<p>

https://github.com/Rectify11/Installer/issues/509
https://github.com/Rectify11/Installer/issues/526
https://github.com/Rectify11/Installer/issues/585
https://github.com/Rectify11/Installer/issues/400
https://github.com/Rectify11/Installer/issues/440

https://github.com/MicaForEveryone/MicaForEveryone/issues/399
https://github.com/MicaForEveryone/MicaForEveryone/issues/386
https://github.com/MicaForEveryone/MicaForEveryone/issues/61
https://github.com/MicaForEveryone/MicaForEveryone/issues/61

https://github.com/Maplespe/ExplorerBlurMica/issues/117
https://github.com/Maplespe/ExplorerBlurMica/issues/136#issuecomment-2074087174

</p>
</details>

**Comment by snowie2000** [comment]
Modifying the alpha bit of texts defies the rule to be fully compatible with what GDI does. 

It will for sure cause some apps to behavior incorrectly. 

Virtual box is one of its kind. It relies on the alpha bit to draw texts on its UI.

**Comment by Undisputed00x** [comment]
Maybe add it as a hidden experimental option in the config file 😅. Blacklisting of conflicting processes can come into play here. Obviously this doesn't address to the average user.
